src/librustc/mir/interpret/allocation.rs

```diff
@@ -688,6 +693,56 @@ impl<Tag> DerefMut for Relocations<Tag> {
     }
 }
 
+/// A partial, owned list of relocations to transfer into another allocation.
+pub struct AllocationRelocations<Tag> {
+    relative_relocations: Vec<(Size, (Tag, AllocId))>,
+}
```

Comment:
Interesting idea. Long-term (does not have to happen in this PR), it might make sense to combine this with the run-length encoded undef mask into one struct? They are AFAIK only used together, and basically together opaquely represent "metadata" (on top of the raw bytes) that memcpy needs to copy. Then there would be two operations, tentatively named `get_metadata` and `apply_metadata` or so.